snapcraft/repo.py

```diff
@@ -58,11 +58,16 @@
 
 
 def install_build_packages(packages):
+    seen = set()
+    seen_add = seen.add
+    unique_packages = [x for x in packages if not (x in seen or seen_add(x))]
     new_packages = []
-    for pkg in packages:
+    for pkg in unique_packages:
         try:
-            if not apt.Cache()[pkg].installed:
+            apt_cache = apt.Cache()
```

Comment:
I wonder if something like this is allowed (apt cache being context aware, I think it is), so mind replacing that with

```python
with apt.Cache() as apt_cache:
    if not apt_cache[pkg].installed:
        new_package.append(pkg)
```

It might also be more performant to open the apt cache before looping over the packages.